src/modm/platform/dma/stm32/dma_base.hpp.in

```diff
@@ -182,10 +232,36 @@ protected:
 			DMA_CCR_CIRC     |						// CircularMode
 			DMA_CCR_PL_1     | DMA_CCR_PL_0;		// Priority
 %% endif
+
+	enum class Interrupt {
+		Global = 0x01,
+		TransferComplete = 0x02,
+		HalfTransferComplete = 0x04,
+		Error = 0x08,
+		All = 0x0f,
+	};
+	MODM_FLAGS32(Interrupt);
+
+	using IrqHandler = void (*)(void);
```

Comment:
https://de.cppreference.com/w/cpp/utility/functional/function

Reply:
Possible, keep in mind that `std::function` will have noticeable overhead in debug builds. Overheads like 5700% to a simple function call have been measured ([here](https://vittorioromeo.info/index/blog/passing_functions_to_functions.html) on x86) because the compiler will not inline its implementation. I don't know how much that actually matters.

Also I am not sure if std::function would still compile when dynamic allocation is disabled, although it should not allocate for simple function pointers due to small object optimization.

Reply:
Oh, thanks for the comment. Staying with `void (*)(void)` sounds like a better idea to me...